com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -20,174 +20,73 @@ public enum GridDepthType { Channel, ChannelHot, Counting };
     /// </summary>
     public class GridSensor : ISensor, IBuiltInSensor
     {
-        /// <summary>
-        /// Name of this grid sensor.
-        /// </summary>
-        string Name;
-
-        //
-        // Main Parameters
-        //
-
-        /// <summary>
-        /// The scale of each grid cell.
-        /// </summary>
-        Vector3 CellScale;
-
-        /// <summary>
-        /// The number of grid on each side.
-        /// </summary>
-        Vector3Int GridNumSide;
-
-        /// <summary>
-        /// Rotate the grid based on the direction the agent is facing.
-        /// </summary>
-        bool RotateWithAgent;
-
-        /// <summary>
-        /// Array holding the depth of each channel.
-        /// </summary>
-        int[] ChannelDepth;
-
-        /// <summary>
-        /// List of tags that are detected.
-        /// </summary>
-        string[] DetectableObjects;
-
-        /// <summary>
-        /// The layer mask.
-        /// </summary>
-        LayerMask ObserveMask;
-
-        /// <summary>
-        /// The data layout that the grid should output.
-        /// </summary>
-        GridDepthType gridDepthType = GridDepthType.Channel;
-
-        /// <summary>
-        /// The reference of the root of the agent. This is used to disambiguate objects with the same tag as the agent. Defaults to current GameObject.
-        /// </summary>
-        GameObject rootReference;
-
-        int MaxColliderBufferSize;
-
-        int InitialColliderBufferSize;
-        Collider[] m_ColliderBuffer;
-
-        float[] m_ChannelBuffer;
-
-        //
-        // Hidden Parameters
-        //
-
-        /// <summary>
-        /// The total number of observations per cell of the grid. Its equivalent to the "channel" on the outgoing tensor.
-        /// </summary>
-        int ObservationPerCell;
-
-        /// <summary>
-        /// The offsets used to specify where within a cell's allotted data, certain observations will be inserted.
-        /// </summary>
-        int[] ChannelOffsets;
+        string m_Name;
+        Vector3 m_CellScale;
+        Vector3Int m_GridNum;
```

Comment:
I explored grouping these parameters into struct. The constructer became cleaner but the code becomes much more verbose since these are used a lot everywhere. Also that brings another problem that in the struct some parameters can be changed at runtime while some cannot, and that will make it even more complicated.
So I decided to leave it as is.